### DISCUSSION TITLE
ECONNREFUSED

### DISCUSSION THREAD

**Discussion by Tky-Phoenix** [ghostfolio/ghostfolio#919] [Q&A]
Hey guys,
I tried following the documentation but the container crashes everytime and I get the below error message. I'm still a beginner and just don't understand how to solve this.

Any advice appreciated, thank you!

[ghostfolio_error_20220514.txt](https://github.com/ghostfolio/ghostfolio/files/8692573/ghostfolio_error_20220514.txt)

**Comment by dtslvr** [comment]
Hi @Tky-Phoenix

Have you set the environment variables? The error could be related to this: https://github.com/ghostfolio/ghostfolio/issues/916#issuecomment-1125951499

**Reply by Tky-Phoenix** [reply]
Thank you @dtslvr !
That was the first problem I encountered. I realized I needed to set the two variables, 
ACCESS_TOKEN_SALT and JWT_SECRET_KEY but I wasn't sure what to set them to. I used a random string like this 8nQM6GhFog6rJ8Mvqi. 

EDIT: I checked the link and fixed the two variables above. That should be good.
Still getting this error though. I understand it is related to REDIS but I'm not sure how to fix it.

[ghostfolio_error_20220514_2.txt](https://github.com/ghostfolio/ghostfolio/files/8692670/ghostfolio_error_20220514_2.txt)


(As you can probably tell, I'm a complete noob and I am using Docker GUI instead of CLI)

**Reply by dtslvr** [reply]
Yes, random strings (secrets) are fine. What about the other environment variables like `REDIS_HOST` and `REDIS_PORT`? Is your _Redis_ container running?

**Reply by Tky-Phoenix** [reply]
Yes, Redis is up and running. I made sure the REDIS_HOST and REDIS_PORT match with the Redis container
![Screen Shot 2022-05-14 at 21 16 01](https://user-images.githubusercontent.com/105540215/168425311-7ebe9b22-7b62-4d79-9769-f0ef4837eaad.png)
.

**Reply by Tky-Phoenix** [reply]
@dtslvr I started from scratch and used the guidance under a. Run environment but get the following errors now. Do you have any idea what might be the cause here and how to fix it? Thank you in advance.

urllib3.exceptions.ProtocolError: ('Connection aborted.', PermissionError(13, 'Permission denied'))
requests.exceptions.ConnectionError: ('Connection aborted.', PermissionError(13, 'Permission denied'))
docker.errors.DockerException: Error while fetching server API version: ('Connection aborted.', PermissionError(13, 'Permission denied'))
[19011] Failed to execute script docker-compose

**Reply by dtslvr** [reply]
`Permission denied` may be because you do not have root privileges. Either run the commands with `sudo ...` or set up a docker group: https://stackoverflow.com/a/48957722/744012

**Reply by Tky-Phoenix** [reply]
@dtslvr Thank you, that was it! I did it with sudo now and it worked at least for the first step. On the second step I get a 

Error: P1001: Can't reach database server at `postgres`:`5432`

Please make sure your database server is running at `postgres`:`5432`.
error Command failed with exit code 1.

I checked the postgres container and the port settings are correct. I am not sure why it cannot reach it.
